tests/test_de.py

```diff
@@ -0,0 +1,123 @@
+""" test_de.py
+    integration test to verify that de process generates expected output
+"""
+
+import unittest
+import sys
+import hashlib
+import os
+import pandas as pd
+
+sys.path.append("../ingest")
+from cell_metadata import CellMetadata
+from clusters import Clusters
+from de import DifferentialExpression
+
+
+def get_annotation_labels(metadata, annotation, de_cells):
+    """ extract annotation_labels for specified annotation
+        from metadata file filtered by cluster file cells
+    """
+    dtypes = DifferentialExpression.determine_dtypes(
+        metadata.headers, metadata.annot_types
+    )
+    de_annots = DifferentialExpression.subset_annots(metadata, de_cells)
+    unique_labels = de_annots[annotation].unique()
+    return unique_labels.tolist()
+
+
+def find_expected_files(labels, cluster_name, annotation, method):
+    """ Check that files were created for all expected annotation labels
+    """
+    found = 0
+    for label in labels:
+        sanitized_label = label.replace(" ", "_")
+        expected_file = (
+            f"{cluster_name}--{annotation}--{str(sanitized_label)}--{method}.tsv"
```

Comment:
`str` seems extraneous; `replace` assumes string, and f-string interpolation coerces to string.

```suggestion
        expected_file = (
            f"{cluster_name}--{annotation}--{sanitized_label}--{method}.tsv"
```

Reply:
Good point. The string coercion was put in place before I started sanitizing the label so any coercion is needed before the sanitizing step. So I need to step back and think about what is needed if a cell is not annotated with cell_type__ontology_label (cell_type is optional metadata so this is reasonable in our validated metadata). For those cases the label would be NaN which is a float (and would be represented in visualizations as "Unspecified").

Reply:
I work on addressing the above issue in`jlc_sparse_de` to avoid more "artifactual merge commits"